go-controller/pkg/node/node_ip_handler_linux.go

```diff
@@ -62,25 +63,44 @@ func (c *addressManager) delAddr(ip net.IP) bool {
 }
 
 func (c *addressManager) Run(stopChan <-chan struct{}) {
-	errorCallback := func(err error) {
-		klog.Errorf("Failed during AddrSubscribe callback: %v", err)
-	}
+	var addrChan chan netlink.AddrUpdate
 	addrSubscribeOptions := netlink.AddrSubscribeOptions{
-		ErrorCallback: errorCallback,
+		ErrorCallback: func(err error) {
+			klog.Errorf("Failed during AddrSubscribe callback: %v. Calling sync() explicitly", err)
+			// sync the manager with current addresses on the node
+			c.sync()
+		},
 	}
-	addrChan := make(chan netlink.AddrUpdate)
-	if err := netlink.AddrSubscribeWithOptions(addrChan, stopChan, addrSubscribeOptions); err != nil {
-		klog.Errorf("Unable to run Node IP Manager, error during netlink subscribe: %v", err)
-		return
+
+	subScribeFcn := func() error {
+		addrChan = make(chan netlink.AddrUpdate)
+		if err := netlink.AddrSubscribeWithOptions(addrChan, stopChan, addrSubscribeOptions); err != nil {
+			return err
+		}
+		// sync the manager with current addresses on the node
+		c.sync()
```

Comment:
@trozet @astoycos this sync may now be a bit redundant, since we will be calling sync in the ticker timer.
May we remove it?

Reply:
Well if we remove that sync we will never sync the existing addresses in the standard start scenario (and instead only do it 30 seconds later and so on) so I think we still need it even if it's a little bit redundant.  

I also think adding the ticker to manually `sync()` every 30 seconds kind of defeats the purpose of subscribing to the netlink addr channel at all right?

I thought the goal was to 
1. Resubscribe and sync addresses on any channel error 
2. Manually sync 30 seconds after any "failure + resubscribe" event occur  

We're now manually calling sync() every 30 seconds right regardless right?

Reply:
Good point on if we removed sync() from `subScribeFcn`, we would nee to wait 30 secs for the periodic timer.
It feels redundant for sure to call sync every 30 seconds. @trozet is that what you are expecting?

Reply:
I wanted the 30 second safety timer to only fire if we havent received any update from netlink in 30 seconds. It looks like that is what Flavio has now as any hit in the select where we receive something off the channel or it closes will result in resetting the dead timer.